pandas/core/indexes/period.py

```diff
@@ -645,7 +645,8 @@ def start_time(self):
 
     @property
     def end_time(self):
-        return self.to_timestamp(how='end')
+        data = (self + 1).start_time.values.astype(int) - 1
```

Comment:
this is reaching into the internals too much
is to_timestakp working? (is it tested)?
fix should be there

Reply:
Ok I've just pushed a new commit where I tried to make the changes in ``to_timestamp`` rather than in ``end_time``. 

It seems to work as expected but is causing a few tests to fail because the tests expected the time part to be 00:00:00. For example:
```
In [1]: pi = pd.PeriodIndex(['2011-01', '2011-02', '2011-03'], freq='M')
In [2]: exp = pd.DatetimeIndex(['2011-01-31', '2011-02-28', '2011-03-31'])

In [3]: pi.astype('datetime64[ns]', how='end')[0])
Out[3]: 2011-01-31 23:59:59.999999999

In [4]: exp[0]
Out[4]: 2011-01-31 00:00:00
```
Previously these were equal but now [3] has a time of 23:59:59.999999999 and [4] has a time of 00:00:00.